go-controller/pkg/ovn/ovn.go

```diff
@@ -114,16 +113,11 @@ type Controller struct {
 	// An address set factory that creates address sets
 	addressSetFactory addressset.AddressSetFactory
 
-	// For each logical port, the number of network policies that want
-	// to add an ingress deny rule.
-	lspIngressDenyCache map[string]int
-
-	// For each logical port, the number of network policies that want
-	// to add an egress deny rule.
-	lspEgressDenyCache map[string]int
-
-	// A mutex for lspIngressDenyCache and lspEgressDenyCache
-	lspMutex *sync.Mutex
+	// map of existing shared port groups for network policies
+	// port group exists in the db if and only if port group key is present in this map
+	// key is namespace
+	// make sure to not take any other Lock while holding sharedNetpolPortGroups key lock
```

Comment:
when I use more that 1 lock, I need to make sure there is no deadlock possible, and that the required order of locking is always the same.
e.g. network policy will acquire `sharedNetpolPortGroups` while holding networkPolicy own lock. In that case I will leave comment that `sharedNetpolPortGroups` is not expect to take any other locks, and I will leave comment around networkPolicy lock that it is allowed to take `sharedNetpolPortGroups` lock, but not e.g. namespaceInfo Lock.
So this comment literally means don't take ANY other lock while holding this. Then I can understand if it is safe to take some lock by reading its comment.

Reply:
still it seems a bit ambiguous. There are tons of different locks in ovnk8s and this comment is added in the overall OVN controller. I'd like to see specifically which locks you are concerned about as a comment here (np lock, etc). Someone could come along later and add some other lower level lock that is perfectly fine to lock with. At least if you added the locks you are thinking about specifically here it would give a reader some insight into the deadlock interaction you have already thought of.